cmake/externals/opencv.cmake

```diff
@@ -37,7 +37,7 @@ set(BUILD_TBB                   OFF CACHE INTERNAL "")
 set(BUILD_TIFF                  OFF CACHE INTERNAL "")
 set(BUILD_WEBP                  OFF CACHE INTERNAL "")
 set(BUILD_WITH_STATIC_CRT       OFF CACHE INTERNAL "")
-if(IOS)
+if(APPLE)

```

Comment:
enabling openCV to build zlib from source for macOS as well otherwise will result in undefined symbols issue in the output binary.

but `APPLE` covers more general apple platforms (including watchos, tvos, etc). not sure if there's a narrower definition could be used here.